BUG: test_cython Failing on Windows on ARM64 with clang-cl

### Describe your issue.

- The scipy.special.tests.test_extending test suite is failing on Windows ARM64 when built using clang-cl and flang-new. However, the same test passes when built with MSVC (cl). The failure occurs in the cy_gamma function, which calls gamma(z) from scipy.special.cython_special.

- I ran the same test with different inputs, but each time it returned the same incorrect value: 6.95307592837223e-310+0j

- However, when I added print statements inside the source code, it displayed the correct value, but the returned result was still incorrect.

### Reproducing Code Example

```python
Build SciPy from Source on Windows on ARM64. 
Run the test using this command 
python dev.py test -t scipy.special.tests.test_extending
```

### Error message

```shell
scipy\special\tests\test_extending.py:24: in test_cython  
assert extensions_cpp.cy_gamma(0.5 + 1.0j) == gamma(0.5 + 1.0j)         
E   AssertionError: assert (6.9530677558909e-310+0j) == np.complex128(0.300694617260656-0.4249678794331237j) 
E    +  where (6.9530677558909e-310+0j) = <cyfunction cy_gamma at 0x0000015A11B32670>((0.5 + 1j))             
E    +    where <cyfunction cy_gamma at 0x0000015A11B32670> = <module 'extending_cpp' from 'C:\\Users\\HCKTest\\AppData\\Local\\Temp\\pytest-of-HCKTest\\pytest-33\\test_cython0\\special\\tests\\_cython_examples\\build\\extending_cpp.cp312-win_arm64.pyd'>.cy_gamma    
 E    +  and   np.complex128(0.300694617260656-0.4249678794331237j) = gamma((0.5 + 1j))           
extensions = <module 'extending' from 'C:\\Users\\HCKTest\\AppData\\Local\\Temp\\pytest-of-HCKTest\\pytest-33\\test_cython0\\special\\tests\\_cython_examples\\build\\extending.cp312-win_arm64.pyd'>   
extensions_cpp = <module 'extending_cpp' from 'C:\\Users\\HCKTest\\AppData\\Local\\Temp\\pytest-of-HCKTest\\pytest-33\\test_cython0\\special\\tests\\_cython_examples\\build\\extending_cpp.cp312-win_arm64.pyd'>   
srcdir     = 'C:\\Pratham\\scipy\\build-install\\Lib\\site-packages\\scipy\\special'    
tmp_path   = WindowsPath('C:/Users/HCKTest/AppData/Local/Temp/pytest-of-HCKTest/pytest-33/test_cython0')                                                                                                                      
```

### SciPy/NumPy/Python version and system information

<details>

```shell

  "Compilers": {
    "c": {
      "name": "clang-cl",
      "linker": "lld-link",
      "version": "19.1.5",
      "commands": "clang-cl"
    },
    "cython": {
      "name": "cython",
      "linker": "cython",
      "version": "3.0.12",
      "commands": "cython"
    },
    "c++": {
      "name": "clang-cl",
      "linker": "lld-link",
      "version": "19.1.5",
      "commands": "clang-cl"
    },
    "fortran": {
      "name": "llvm-flang",
      "linker": "lld-link",
      "version": "19.1.5",
      "commands": "flang-new"
    },
    "pythran": {
      "version": "0.17.0",
      "include directory": "..\\venv\\Lib\\site-packages\\pythran"
    }
  },
  "Machine Information": {
    "host": {
      "cpu": "aarch64",
      "family": "aarch64",
      "endian": "little",
      "system": "windows"
    },
    "build": {
      "cpu": "aarch64",
      "family": "aarch64",
      "endian": "little",
      "system": "windows"
    },
    "cross-compiled": false
  },
  "Build Dependencies": {
    "blas": {
      "name": "openblas",
      "found": true,
      "version": "0.3.28",
      "detection method": "pkgconfig",
      "include directory": "C:/openblas/include",
      "lib directory": "C:/openblas",
      "openblas configuration": "USE_64BITINT= NO_CBLAS= NO_LAPACK= NO_LAPACKE= DYNAMIC_ARCH=OFF DYNAMIC_OLDER=OFF NO_AFFINITY=1 USE_OPENMP= ARMV8 MAX_THREADS=22",
      "pc file directory": "C:/openblas"
    },
    "lapack": {
      "name": "openblas",
      "found": true,
      "version": "0.3.28",
      "detection method": "pkgconfig",
      "include directory": "C:/openblas/include",
      "lib directory": "C:/openblas",
      "openblas configuration": "USE_64BITINT= NO_CBLAS= NO_LAPACK= NO_LAPACKE= DYNAMIC_ARCH=OFF DYNAMIC_OLDER=OFF NO_AFFINITY=1 USE_OPENMP= ARMV8 MAX_THREADS=22",
      "pc file directory": "C:/openblas"
    },
    "pybind11": {
      "name": "pybind11",
      "version": "2.13.6",
      "detection method": "config-tool",
      "include directory": "unknown"
    }
  },
  "Python Information": {
    "path": "C:\\Pratham\\scipy\\venv\\Scripts\\python.exe",
    "version": "3.12"
  }
}  
```

</details>

--- a/scipy/special/tests/_cython_examples/meson.build
+++ b/scipy/special/tests/_cython_examples/meson.build
@@ -20,6 +20,7 @@ py3.extension_module(
   'extending.pyx',
   install: false,
   cython_args: cython_args,
+  c_args: ['-DCYTHON_CCOMPLEX=0'] # see gh-18975 for why we need this
 )
 
 extending_cpp = fs.copyfile('extending.pyx', 'extending_cpp.pyx')
@@ -29,4 +30,5 @@ py3.extension_module(
   install: false,
   override_options : ['cython_language=cpp'],
   cython_args: cython_args,
+  cpp_args: ['-DCYTHON_CCOMPLEX=0']
 )

